.github/workflows/protobuf-dockerimage.yml

```
@@ -10,25 +10,55 @@ on:
       - protobuf/Dockerfile
       - protobuf/protoc-wrapper
 
+env:
+  PLATFORMS: "linux/arm, linux/arm64, linux/amd64"
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - name: Build the Docker image
-      run: docker build protobuf/. -t build-protobuf
-    - name: Push the Docker image
-      if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/')
-      run: |
-        echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
-        function tag_and_push {
-          docker tag build-protobuf "otel/build-protobuf:${1}" && docker push "otel/build-protobuf:${1}"
-        }
-        if [[ "${GITHUB_REF}" == "refs/heads/main" ]]; then
-          tag_and_push "latest"
-        elif [[ "${GITHUB_REF}" =~ refs/tags/v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
-            TAG="${GITHUB_REF#"refs/tags/v"}"
-            tag_and_push "${TAG}"
-        else
-          tag_and_push "${GITHUB_REF#"refs/tags/"}"
-        fi
\ No newline at end of file
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: |
+          name=otel/build-protobuf
+        tags: |
+          type=edge
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+
+    - name: Setup QEMU
+      uses: docker/setup-qemu-action@v2
+      with:
+        platforms: arm,arm64
+      # if: github.event_name != 'pull_request' # remove after PR testing
+
+    - name: Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: DockerHub Login
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+        if: github.event_name != 'pull_request'
```

```suggestion
      if: github.event_name != 'pull_request'
```
https://github.com/open-telemetry/build-tools/actions/runs/3855989360/jobs/6665721857